substrate/frame/scheduler/src/lib.rs

```diff
@@ -1137,18 +1277,31 @@ impl<T: Config> Pallet<T> {
 					}
 					let wake = now.saturating_add(period);
 					match Self::place_task(wake, task) {
-						Ok(_) => {},
+						Ok(new_address) => {
+							if let Some(RetryConfig { total_retries, remaining, period }) =
+								Retries::<T>::take((when, agenda_index))
+							{
+								// Reset the counter if the task ran successfully.
+								let remaining = if failed { remaining } else { total_retries };
```

Comment:
Doc comment is outdated, I'll fix it. This is not needed anymore since the impl changed to retry clones. It had no effect on the current impl because for original tasks, `remaining == total_retries` anyway and retry clones don't get to this part of the code because they are not "periodic".

Reply:
Done.